tests/utils/test_client.py

```diff
@@ -53,4 +58,33 @@ def test_client_enforce_healthy():
     assert my_client.port == int(CONFIG["my_instance_name"]["port"])
     assert my_client.protocol == CONFIG["my_instance_name"]["protocol"]
     assert my_client.base_path == "/api/versioned/v1/"
-    assert my_client.auth == UsernamePasswordAuth("admin", os.environ["TAMR_TOOLBOX_PASSWORD"],)
+    assert my_client.auth == UsernamePasswordAuth(
+        "admin", os.environ["TAMR_TOOLBOX_PASSWORD"],
+    )
+
+
+@mock_api()
+def test_get_with_connection_retry():
+    # To ensure timeout error is raised to test correct ConnectionError is caught
+
+    log_prefix = "caught_connection_error"
+    log_file_path = os.path.join(
+        tempfile.gettempdir(), f"{log_prefix}_{utils.logger._get_log_filename()}"
+    )
+    log_file_path_extended = os.path.join(
+        log_file_path, utils.logger._get_log_filename()
+    )  # because gettempdir creates a directory
+    if os.path.exists(log_file_path_extended):
+        os.remove(log_file_path_extended)
+
+    my_client = utils.client.create(**CONFIG["my_other_instance"])
+    utils.logger.enable_toolbox_logging(
+        log_to_terminal=False, log_directory=log_file_path
+    )
+
+    with pytest.raises(TimeoutError):
+        utils.client.get_with_connection_retry(
+            my_client, "/api/service/health", timeout_seconds=10, sleep_seconds=1
+        )
+
+    assert "ConnectionError" not in open(log_file_path_extended).read()
```

Comment:
You actually want to confirm that a ConnectError did occur, but that it was caught and the intended log message was written.  Also, it is best practice to use a context manager (`with ... as ...:`) when reading a file:
```suggestion
    with open(log_file_path_extended, "r") as f:
        # confirm that the intended warning was written to the log
        assert "Caught exception in connect" in f.read()
```